statsmodels/stats/proportion.py

```diff
@@ -188,25 +193,40 @@ def _check(x: np.ndarray, name: str) -> np.ndarray:
         nobs_a = _check(np.asarray(nobs_a), "count")
 
     q_ = count_a / nobs_a
-    alpha_2 = 0.5 * alpha
+
+
+    if alternative == 'two-sided':
+        if method != "binom_test":
+            alpha = alpha / 2.0
+    elif alternative not in ['smaller', 'larger']:
+        raise NotImplementedError(f"alternative {alternative} is not available")
 
     if method == "normal":
         std_ = np.sqrt(q_ * (1 - q_) / nobs_a)
-        dist = stats.norm.isf(alpha / 2.0) * std_
+        dist = stats.norm.isf(alpha) * std_
         ci_low = q_ - dist
         ci_upp = q_ + dist
-    elif method == "binom_test":
-        # inverting the binomial test
+    elif method == "binom_test" and alternative == 'two-sided':
         def func_factory(count: int, nobs: int) -> Callable[[float], float]:
             if hasattr(stats, "binomtest"):
-
                 def func(qi):
-                    return stats.binomtest(count, nobs, p=qi).pvalue - alpha
-
+                    if alternative == 'two-sided':
+                        print(qi, stats.binomtest(count, nobs, p=qi).pvalue - alpha)
+                        return stats.binomtest(count, nobs, p=qi).pvalue - alpha
+                    elif alternative == 'smaller':
```

Comment:
AFAICS, these will never be used because of the if above.
so we don't need it

Reply:
I removed it from the latest commit.